drivers/hmis/app/graphql/mutations/delete_client.rb

```diff
@@ -0,0 +1,74 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class DeleteClient < BaseMutation
+    argument :id, ID, required: true
+    argument :confirmed, Boolean, 'Whether warnings have been confirmed', required: false
+
+    field :client, Types::HmisSchema::Client, null: true
+
+    def resolve(id:, confirmed: false)
+      client = Hmis::Hud::Client.find_by(id: id)
+
+      warnings, resolvable_enrollments = check_enrollments(client, ignore_warnings: confirmed)
+
+      return { client: nil, errors: warnings } if warnings.present?
```

Comment:
(optional nitpick) I prefer to use `any?` for arrays. Doesn't look like it would cause any issue here, but for example `[nil].present?` returns true.
```suggestion
    return { errors: warnings } if warnings.any?
```

Reply:
If `check_enrollments` returns an HmisError:Errors object, you can also call `any?` on that and return it directly.

Reply:
Ah, cool, I wasn't aware `any?` was that smart. I'll probably use that for more things going forward